protobuf/lib/src/protobuf/extension_registry.dart

```diff
@@ -104,14 +107,58 @@ T _reparseMessage<T extends GeneratedMessage>(
         unknownField.writeTo(tagNumber, codedBufferWriter);
       }
       resultUnknownFields._fields.remove(tagNumber);
+      reparsedExtensions = true;
```

Comment:
I guess we should only do this (and I guess the line above...) if `unknownField != null`